src/platform_impl/macos/app_delegate.rs

```diff
@@ -120,14 +134,66 @@ extern "C" fn application_open_urls(_: &Object, _: Sel, _: id, urls: &NSArray<NS
 
   let urls = unsafe {
     (0..urls.count())
-      .flat_map(|i| url::Url::parse(&urls.objectAtIndex(i).absoluteString().unwrap().to_string()))
+      .flat_map(|i| {
+        url::Url::parse(
+          &urls
+            .objectAtIndex(i)
+            .absoluteString()
+            .map(|s| s.to_string())
+            .unwrap_or("".to_owned()), // empty string is not a link so gets ignored / can't crash
+        )
+      })
```

Comment:
this change is unrelated, was just trying to debug a startup crash in my application at the same time, not sure yet if the issue was even this unwrap.

I'll probably remove it before marking this pr as ready to review.